mypy/nodes.py

```diff
@@ -1125,6 +1126,7 @@ def __init__(
         # Used for error reporting (to keep backwad compatibility with pre-3.8)
         self.deco_line: int | None = None
         self.removed_statements = []
+        self.dataclass_transform_spec: DataclassTransformSpec | None = None
```

Comment:
Can you add this to `TypeInfo` instead? We've tried to keep `ClassDef` pretty simple and move most meta-information to `TypeInfo`.